src/libsyntax_pos/lib.rs

```diff
@@ -227,14 +227,6 @@ impl SpanData {
     }
 }
 
-// The interner is pointed to by a thread local value which is only set on the main thread
-// with parallelization is disabled. So we don't allow `Span` to transfer between threads
-// to avoid panics and other errors, even though it would be memory safe to do so.
-#[cfg(not(parallel_compiler))]
-impl !Send for Span {}
-#[cfg(not(parallel_compiler))]
-impl !Sync for Span {}
```

Comment:
Why are those removed? This wasnt a hack, but it ensured that `Span` is accessed from another thread, which would have a different span interner (or none). It is a logic bug to access a `Span` with a different span interner.

Reply:
Er, yes, that was a mistake, thanks.

Reply:
(I assume I should not `r+` until these are put back...)